test/extended/storage/driver_configuration.go

```diff
@@ -0,0 +1,399 @@
+	}
+
+	clusterCSIDriver.Spec.DriverConfig.VSphere = clusterCSIDriverOptions
+
+	_, err = oc.AdminOperatorClient().OperatorV1().ClusterCSIDrivers().Update(ctx, clusterCSIDriver, metav1.UpdateOptions{})
+	if err != nil {
+		e2e.Failf("failed to update ClusterCSIDriver: %v", err)
+	}
+}
+
+func loadAndCheckCloudConf(ctx context.Context, oc *exutil.CLI, sectionName string, cloudConfigOptions map[string]string) error {
+	e2e.Logf("Validating cloud.conf section %s", sectionName)
+
+	cm, err := oc.AdminKubeClient().CoreV1().ConfigMaps("openshift-cluster-csi-drivers").Get(ctx, "vsphere-csi-config", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get ConfigMap: %v", err)
+	}
+
+	cloudConfData, ok := cm.Data["cloud.conf"]
+	if !ok {
+		return fmt.Errorf("cloud.conf key not found in ConfigMap")
+	}
+
+	cfg, err := ini.Load([]byte(cloudConfData))
+	if err != nil {
+		return fmt.Errorf("failed to load cloud.conf: %v", err)
+	}
+
+	section, err := cfg.GetSection(sectionName)
+	if err != nil {
+		return fmt.Errorf("section %s not found in cloud.conf: %v", sectionName, err)
+	}
+
+	if !reflect.DeepEqual(section.KeysHash(), cloudConfigOptions) {
+		return fmt.Errorf("check of %s section in cloud.conf failed, got: %v expected: %v", sectionName, section.KeysHash(), cloudConfigOptions)
+	}
+
+	e2e.Logf("Validation of %s section in cloud.conf succeeded", sectionName)
+
+	return nil
+}
+
+func validateSnapshotCreation(ctx context.Context, oc *exutil.CLI, successfulSnapshotsCreated int) {
+	e2e.Logf("Validating snapshot creation.")
+
+	pvc, err := createTestPVC(ctx, oc, oc.Namespace(), "test-pvc", "1Gi")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	defer func() {
+		oc.AdminKubeClient().CoreV1().PersistentVolumeClaims(oc.Namespace()).Delete(ctx, pvc.Name, metav1.DeleteOptions{})
+	}()
+
+	pod, err := createTestPod(ctx, oc, pvc.Name, oc.Namespace())
+	o.Expect(err).NotTo(o.HaveOccurred())
+	defer func() { oc.KubeClient().CoreV1().Pods(oc.Namespace()).Delete(ctx, pod.Name, metav1.DeleteOptions{}) }()
+
+	// Wait for pvc to be bound.
+	o.Eventually(func() v1.PersistentVolumeClaimPhase {
+		pvc, err := oc.AdminKubeClient().CoreV1().PersistentVolumeClaims(oc.Namespace()).Get(ctx, "test-pvc", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		return pvc.Status.Phase
+	}).Should(o.Equal(v1.ClaimBound))
+
+	var wg sync.WaitGroup
+	snapshotsCreatedChan := make(chan string, successfulSnapshotsCreated)
+
+	for i := 0; i < successfulSnapshotsCreated; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			snapshotName := fmt.Sprintf("test-snapshot-%d", i)
+			err := createSnapshot(oc, oc.Namespace(), snapshotName, "test-pvc")
+			if err != nil {
+				e2e.Failf("failed to create snapshot: %v", err)
+				return
+			}
+			snapshotsCreatedChan <- snapshotName
+		}(i)
+	}
+	wg.Wait()
+	close(snapshotsCreatedChan)
+
+	var snapshotsCreated = make([]string, 0, successfulSnapshotsCreated)
+	for snapshotName := range snapshotsCreatedChan {
```

Comment:
Isn't the channel closed here ?!

Reply:
I'm not sure why there is a channel with snapshot names and then it's transformed into array of names. Can't you create array of snapshot names directly?

Reply:
Indeed, just wanted a quick check with payload to see if concurrent snapshots work fine and didn't get back to it yet. Corrected.